sdk/python/kfp/components/component_factory.py

```diff
@@ -435,3 +436,117 @@ def create_component_from_func(func: Callable,
 
     return python_component.PythonComponent(
         component_spec=component_spec, python_func=func)
+
+
+def extract_container_component_interface(
```

Comment:
Thanks for this comment! In theory it could, but that would require a refactoring of the extract_component_interface function. The main difference is that we don't parse the return type annotation for container_component early in the function. It would be possible to refactor this function by separating out the logic of return type annotation and determine whether we need it based on the type of the component. Do you think this refactoring is reasonable?